### ISSUE TITLE
Copy/Paste Events only works in BORIS 8.x , but not BORIS 7.x.

### ISSUE THREAD

**Issue by SirWilksBoothDaThird** [olivierfriard/BORIS#775]
BORIS version(s):
7.9.15
7.13.5
7.13.6
7.13.8
7.13.9
8.0.10
8.0.21
8.27.1
8.6.2
+ Other versions.

Note: For testing purposes, many of the tested BORIS versions are portable, I suspected earlier that this may be a cause for what seems to be a clipboard error of some type, but the error persists in installed BORIS 7.x versions also.

Issue:
Copying and pasting events doesn't work in all BORIS versions 7.x. Program either does nothing or throws an error.

Steps to reproduce:
1. Open a project and any observation with at least one media file loaded in Player #1 (I use an *.MP4 file).
2. In "Events" panel, right-click on any event, either in the timestamp field or any other field in the given row for the event. Choose "Copy events".
3. In "Events" panel, right-click anywhere and chose "Paste events".
4. See no result or the following error:
"The clipboard does not contain events! Events must be organized in 5 columns separated by TAB character".

![Boris-v7x-CopyPasteEvents-IssueScreencap001](https://github.com/user-attachments/assets/d8218d9e-3628-4128-b49e-a792034731ee)
![Boris-v7x-CopyPasteEvents-IssueScreencap002](https://github.com/user-attachments/assets/8f09b210-7b81-40be-be70-461c65768777)
![Boris-v7x-CopyPasteEvents-IssueScreencap003](https://github.com/user-attachments/assets/5f586e45-d28c-46fc-971d-a9617dc38f08)

**Comment by SirWilksBoothDaThird** [comment]
Follow-up: The error log doesn't really seem to point me in the right direction for fixing this.
With my limited knowledge, I suspect that the issue is related to the way BORIS exports the event's CSV data to the clipboard or something. 
I don't get what the "5 columns" request is TBH. "Time - Subject - Code - Type - Modifier - Comment" would yield 6 columns of data, no? Maybe this is the problem but I don't know.
BORIS 8.27.1 is a great program BTW! I would use it because it fixes this issue and works great when copying and pasting events. 
BUT, since it made the switch to using an MPV player as its primary media player instead of the awesome VLC player, jumping forward and backward by precisely 1 second -- which is exactly what I require -- has become impossible for some reason. Now, jumping by 1 second jumps forward by 30 frames, or 18 frames, or 26 frames or whatever FPS I used when recording videos. It uses FPS as second increments, which probably requires re-encoding the videos, and I have like a hundred of them, so I would rather drink acid. I've spent hours on these event copy/paste and time-jumping issues and would happily use version 8.27.1 IF, for the love of god, there was an option to please just switch to VLC as the media player. I wouldn't use version 7 if it weren't for its ability to jump by precise second amounts.

Would it be best to request this as a feature? Why fix something that isn't broken and get rid of a perfectly functional player? The MPV-based BORIS 8.x would be wonderful if there was a check box with both the MPV player and VLC player to choose as the primary media player.

The reason I'm going on this BORIS 7.x time-jump tangent is because BORIS 8.x has everything I need but no option to jump by precise whole-second amounts like the VLC-based BORIS 7.x can. Maybe MPV player can also jump by exactly 1 second and I don't know how to configure this, but if this issue was also fixed in BORIS 8.27.1, I would actually close THIS particular issue seeing as my event copy/paste problems would be solved AND I would have a proper way to jump by precisely 1 second and not 0.998 seconds.

**Comment by SirWilksBoothDaThird** [comment]
If anyone has any insight into whether their version of Boris 7.x has a working "copy/paste events" function, please let me know in this issue as I'm interested to hear other users' experiences with this version in particular.

**Comment by olivierfriard** [comment]
Hi,
version 7 is no longer supported, so the "copy/paste events" issue will not be fixed.

Regarding the one-second jump issue, is it possible to access a sample video?

**Comment by SirWilksBoothDaThird** [comment]
Hi Olivier,

Thank you for your reply to my query. Actually, I seem to have fixed the problem, which is why I've also closed the issue a couple of days ago.

It turns out that what I needed to do was to edit the observation, and make sure that Player 1 is loaded with a 30 fps video file. Actually, the fps is something like 30.0000009 or something, but I've replicated the issue a lot with some videos that have 30.01 fps and 18.29 fps and they also have the weird time jump issue where the jumps amount to 1.12 seconds or similar, non-absolute values.

I'm guessing that it's my fault for using videos that I've pulled from tablet and phone camera without any post-production or re-encoding. At least in BORIS 8.27.1, the issue with 1-second time jump seems to be fixed when primary Media Player 1 is loaded with a video file that has been manually re-encoded at 30 fps (I'm unfamiliar with MPV Player do maybe this particular fps is its default fps?). Once I loaded one of the videos that I had encoded myself, it seems to fix the issue every time. I wish that there was some warning about this, since MPV Player seems to be more sensitive than VLC with non-standard videos that have been recorded on whatever us researchers may have on hand, such as a smartphone.

After a bunch of tests, I can confirm that BORIS 8.27.1 with MPV Player just takes whatever fps that is received by primary media player #1 and uses that to define what is "1 second", which is inconvenient because a video with 29.97 fps, for instance, gives a 1.001 second jump, with 5 jumps equaling 5.005 seconds. I'm not sure how to fix this other than to always have a long 30fps video file on hand that I can load into media player 1 and shrink that player down so that I can't see it. That way, I always get that 1 second jump that I need. Perhaps there is an option to rectify this, but I haven't found anything so far.

The copy and paste issue is strangely present in version 8.27.1, even though a row/event can be copied and pasted in notepad but not pasted in the BORIS 8.27.1 observation list, but I can live with that for now.
